jans-cedarling/cedarling/src/authz/token_data.rs

```diff
@@ -18,27 +17,19 @@ pub(crate) struct TokenPayload {
     pub payload: HashMap<String, serde_json::Value>,
 }
 
-impl TokenPayload {
+impl<'a> TokenPayload {
     fn new(payload: HashMap<String, serde_json::Value>) -> Self {
         Self { payload }
     }
 
-    /// Get claim attribute value
-    /// for all types that implements `ClaimGetter<T>`
-    pub fn get_value<T>(&self, key: &str) -> Result<T, GetTokenClaimValue>
-    where
-        TokenPayload: AttributeGetter<T>,
-    {
-        self.get_attribute_value(key)
-    }
-
-    /// Get claim attribute [`RestrictedExpression`]
-    /// for all types that implements `ClaimGetter<T>`
-    pub fn get_expression<T>(&self, key: &str) -> Result<RestrictedExpression, GetTokenClaimValue>
-    where
-        TokenPayload: AttributeGetter<T>,
-    {
-        self.get_attribute_expression(key)
+    pub fn get(&'a self, key: &'a str) -> Result<Payload<'a>, GetTokenClaimValue> {
+        self.payload
+            .get(key)
+            .map(|value| Payload {
+                key: key.to_string(),
+                value,
+            })
+            .ok_or(GetTokenClaimValue::KeyNotFound(key.to_string()))
     }
 }
```

Comment:
You can omit the lifetime for the return value and simplify it like this 
```
impl TokenPayload {
    fn new(payload: HashMap<String, serde_json::Value>) -> Self {
        Self { payload }
    }

    pub fn get(&self, key: &str) -> Result<Payload, GetTokenClaimValue> {
        self.payload
            .get(key)
            .map(|value| Payload {
                key: key.to_string(),
                value,
            })
            .ok_or(GetTokenClaimValue::KeyNotFound(key.to_string()))
    }
}
```

Reply:
Thanks, resolved